src/main/java/com/paloaltonetworks/panorama/api/mapping/DeviceGroupResponse.java

```diff
@@ -23,28 +22,33 @@
 
 @XmlAccessorType(XmlAccessType.FIELD)
 @XmlRootElement(name="response")
-public class DeviceGroupResponse {
-
-	@XmlAttribute(name = "status")
-	private String status;
-	
-	@XmlElement(name="result")
-	private DeviceGroups deviceGroups;
-	
-	public String getStatus(){
-		return this.status;
-	}
-	
-	public void setStatus(String value) {
-		this.status = value;
-	}
-	
-	public DeviceGroups getDeviceGroups(){
-		return this.deviceGroups;
-	}
-	
-	public void setDeviceGroups(DeviceGroups value){
-		this.deviceGroups = value;
-	}
-	
+public class DeviceGroupResponse implements PANResponse {
+
+    @XmlAttribute(name = "status")
+    private String status;
+
+    @XmlElement(name="result")
+    private DeviceGroups deviceGroups;
+
+    @Override
+    public String getStatus(){
+        return this.status;
+    }
+
+    public void setStatus(String value) {
+        this.status = value;
+    }
+
+    public DeviceGroups getDeviceGroups(){
+        return this.deviceGroups;
+    }
+
+    public void setDeviceGroups(DeviceGroups value){
+        this.deviceGroups = value;
+    }
+
+    @Override
+    public String getCode() {
```

Comment:
i see a lot of places where getCode throws an exception. maybe we should split the interface?

Generally classes implementing interfaces should not throw exceptions for methods called on it unless the interface declares the exception type.

> False unification of interfaces. An interface has two implementors. One implementor implements half of the interface and throws UnsupportedOperationException for the other half. The second implementor implements the other half and throws UnsupportedOperationException for the first half. It would be better to split the interface into two. Similar to RefusedBequest.